### PR TITLE
Docs: Adds local mender-configure artifacts generation

### PR DESCRIPTION
**Pull request by MuchoLucho**
# External Contributor Checklist

🚨 Please review the [guidelines for contributing](https://github.com/mendersoftware/mender/blob/master/CONTRIBUTING.md) to this repository.

- [x] Make sure that all commits follow the conventional commit [specification](https://github.com/mendersoftware/mendertesting/blob/master/commitlint/grammar.md) for the Mender project.

The majority of our contributions are fixes, which means your commit should have
the form below:

```
fix: <SHORT DESCRIPTION OF FIX>

<OPTIONAL LONGER DESCRIPTION>

Changelog: <USER-FRIENDLY-CHANGE-DESCRIPTION> or <None>
Ticket: <TICKET NUMBER> or <None>
```

- [x] Make sure that all commits are signed with [`git --signoff`](https://git-scm.com/book/en/v2/Git-Tools-Signing-Your-Work). Also note that the signoff author must match the author of the commit.

### Description

Please describe your pull request.

Thank you!

### CODE REVIEW

File: 09.Add-ons/10.Configure/docs.md [hunk 1]
```diff
@@ -35,3 +35,38 @@ There are no special device requirements for using Mender Configure, but editing
 from the UI is not supported if the device is using **signed Artifacts**. It is still possible to
 see reported configuration the UI. Configure Artifacts can be
 deployed with a manually signed Artifact, by following the regular Artifact deployment flow.
+
+## How to deploy a configuration to multiple devices at once
+
+The `mender-configure` add-on was designed thinking on being executed from the UI for one device at a time, however, you can generate the artifact manually and then create deployments to groups (static or dynamic) taking advantage of the `mender-configure` Update Module:
```
Comment (kacf) [inline 09.Add-ons/10.Configure/docs.md:41]: Maybe rephrase:

```
The `mender-configure` add-on was designed to be executed from the UI for one device at a time
```

File: 09.Add-ons/10.Configure/docs.md [hunk 2]
```diff
@@ -35,3 +35,38 @@ There are no special device requirements for using Mender Configure, but editing
...
+
+### Prerequisite
+
+You will need an script already localted in `/usr/lib/mender-configure/apply-device-config.d` as described in ["How to make an apply-device-config script"](01.Device-integration/docs.md#how-to-write-an-apply-device-config-script) section of these docs.
```
Comment (kacf) [inline 09.Add-ons/10.Configure/docs.md:45]: "You will need **a** script already **located** in..."

File: 09.Add-ons/10.Configure/docs.md [hunk 3]
```diff
@@ -35,3 +35,38 @@ There are no special device requirements for using Mender Configure, but editing
...
+    -T "mender-configure" \
+    -o "<an-artifact-name>.mender" \
+    -n "<an-artifact-name>" \
+    -t "<your-device-type>" \
+    --software-version "<1>" \
+    --software-filesystem "data-partition" \
+    -m "device-config.json"
+```
+
+Then just upload it to the Mender Server and you are ready to create the deployment for groups of devices.
```
Comment (kacf) [inline 09.Add-ons/10.Configure/docs.md:72]: Maybe also mention that you need to deploy from the normal Deployment section of the UI, not from the Configure section. It should be obvious, since Configure is per-device, but just to make it super clear.

**Review by kacf — APPROVED** [review]
Looks good, just a couple of minor corrections from me. I'm approving now so you can merge immediately after fixing.

File: 09.Add-ons/10.Configure/docs.md (hunk 1 above)
Comment (kacf) [reply 09.Add-ons/10.Configure/docs.md:41]: I think this should fix the passive voice checker as well.

File: 09.Add-ons/10.Configure/docs.md [hunk 4]
```diff
@@ -35,3 +35,38 @@ There are no special device requirements for using Mender Configure, but editing
...
+}
+```
+
+2. Create an artifact with the following command, replace all the data between `<angle-brackets>`.
+
+!!!!! Please note the `-m` flag points to the previously created file, this adds the json as metadata which is expected by the `mender-configure` Update Module on the device side.
+
+```bash
+mender-artifact write module-image  \
+    -T "mender-configure" \
```
Comment (TheMeaningfulEngineer) [inline 09.Add-ons/10.Configure/docs.md:63]: Cool!
So device type `mender-configure` combined with the configuration key value pairs as json in the meta data is the 'configuration format'?

Comment (TheMeaningfulEngineer) [reply 09.Add-ons/10.Configure/docs.md:63]: "T" is the update module name, not device type.

File: 09.Add-ons/10.Configure/docs.md (hunk 1 above)
Comment (MuchoLucho) [reply 09.Add-ons/10.Configure/docs.md]: Handled in 77fac55c119ebb15acf141faf240312c4e841f2b

File: 09.Add-ons/10.Configure/docs.md (hunk 2 above)
Comment (MuchoLucho) [reply 09.Add-ons/10.Configure/docs.md]: Handled in 77fac55c119ebb15acf141faf240312c4e841f2b

File: 09.Add-ons/10.Configure/docs.md (hunk 3 above)
Comment (MuchoLucho) [reply 09.Add-ons/10.Configure/docs.md]: Handled in 77fac55c119ebb15acf141faf240312c4e841f2b

**Comment by MuchoLucho** [comment]
Hello folks, I added a new part of "signed mender-configure" artifacts. Can any of you give me your blessing?

File: 09.Add-ons/10.Configure/docs.md [hunk 5]
```diff
@@ -34,15 +34,15 @@ The picture below shows the configuration update flow.
...
 
 ## How to deploy a configuration to multiple devices at once
 
-The `mender-configure` add-on was designed thinking on being executed from the UI for one device at a time, however, you can generate the artifact manually and then create deployments to groups (static or dynamic) taking advantage of the `mender-configure` Update Module:
+The `mender-configure` add-on was designed to be executed from the UI for one device at a time however, you can generate the artifact manually and then create deployments to groups (static or dynamic) taking advantage of the `mender-configure` Update Module:
 
 ### Prerequisite
 
-You will need an script already localted in `/usr/lib/mender-configure/apply-device-config.d` as described in ["How to make an apply-device-config script"](01.Device-integration/docs.md#how-to-write-an-apply-device-config-script) section of these docs.
+You will need a script already localted in `/usr/lib/mender-configure/apply-device-config.d` as described in ["How to make an apply-device-config script"](01.Device-integration/docs.md#how-to-write-an-apply-device-config-script) section of these docs.
```
Comment (kacf) [inline 09.Add-ons/10.Configure/docs.md:45]: `localted` -> `located`

File: 09.Add-ons/10.Configure/docs.md [hunk 6]
```diff
@@ -69,4 +69,22 @@ mender-artifact write module-image  \
     -m "device-config.json"
 ```
 
-Then just upload it to the Mender Server and you are ready to create the deployment for groups of devices.
+Then just upload it to the _Releases_ section of the Mender Server and you are ready to create the deployment in the _Deployment_ UI for groups of devices.
+
+#### Handle signed Artifacts
+
+You can take advantage of this approach to handle signed artifacts as you can include signatures from your workstation as opposed by using the Mender Configure UI per device to deploy a configuration. Just generate they key and add the `-k private.key` flag into the previous command as described in the [Artifact Signing section](../../06.Artifact-creation/07.Sign-and-verify/docs.md#signing).
```
Comment (kacf) [inline 09.Add-ons/10.Configure/docs.md:76]: "...generate **the** key..."

**Review by kacf — APPROVED** [review]
Looks good!

**Review by TheMeaningfulEngineer — CHANGES_REQUESTED** [review]
Nice!
One thing that is missing is how will the UI behave once you deploy the configuration this way.

The reason why this isn't fully equivalent to using the UI is that this way only the `reported` (of the [configuration API](https://docs.mender.io/api/#management-api-device-configure-get-device-configuration) get changed while the `configure` remains the same.
If you use the API the `configured` changes first and that triggers the generation of the new update.

image

You should confirm that the UI will function normally after using this apprach and document that.

**Comment by MuchoLucho** [comment]
image
I can confirm after usin this approach the UI reflects the change made by using the Artifact instead of using the UI itself

**Comment by TheMeaningfulEngineer** [comment]
@lluiscampos @MuchoLucho 
With the latest discussion where we shouldn't add things to the docs that aren't supported, I'd vote for moving this to mender hub instead.
The configure feature has been developed to work with one device only for the moment.
This is a workaround how to make it work with two.
Seems like ti works ok, but no guarantee that this will be the case in the future.

So I vote for mender hub instead the official docs.

**Comment by lluiscampos** [comment]
FYI @estenberg as docs owner,

> So I vote for mender hub instead the official docs.

Me too :raising_hand_man: 

Closing the PR. @MuchoLucho If capacity permits, follow-up in Mender Hub :innocent:

**Comment by estenberg** [comment]
This is useful, but agree it's fair to have in MH instead since we don't conveniently support multi-device-updates to configure.

However, @MuchoLucho I think a better approach could be instead of creating the Artifacts you just make a script that calls the deviceconfig API to change a configuration attribute? I can take a list of Device IDs and a JSON (or key values).
